libs/jsonforms-components/src/lib/util/inputControlUtils.ts

```diff
@@ -18,3 +21,175 @@ export const isRequiredAndHasNoData = (props: ControlProps) => {
+
+  if (isNotKeyPressTabOrShift(key)) {
+    value = standardizeDate(value) || '';
+    handleChange(path, value);
+  }
+};
+
+/**
+ * Helper function to process for onBlur event for Date/Date Time controls
+ * @param props - EventBlurControlProps
+ */
+export const onBlurForDateControl = (props: EventBlurControlProps) => {
+  const { controlProps } = props;
+  let { value } = props;
+  const { handleChange, path } = controlProps;
+
+  if (isRequiredAndHasNoData(controlProps)) {
+    value = standardizeDate(value) || '';
+    handleChange(path, value);
+  }
+};
+
+/**
+ * Helper function to process onKeyPress events for Time controls
+ * @param props - EventKeyPressControlProps
+ */
+export const onKeyPressForTimeControl = (props: EventKeyPressControlProps) => {
+  const { controlProps, key } = props;
+  const { value } = props;
+  const { handleChange, path } = controlProps;
+
+  if (isNotKeyPressTabOrShift(key)) {
+    handleChange(path, value);
+  }
+};
+
+/**
+ * Helper function to process for onBlur event for time controls
+ * @param props - EventBlurControlProps
+ */
+export const onBlurForTimeControl = (props: EventBlurControlProps) => {
+  const { controlProps } = props;
+  const { value } = props;
+  const { handleChange, path } = controlProps;
+
+  if (isRequiredAndHasNoData(controlProps)) {
+    handleChange(path, value);
+  }
+};
+
+/**
+ * Helper function to process onChange event for Date controls.
+ * @param props - EventChangeControlProps
+ */
+export const onChangeForDateControl = (props: EventChangeControlProps) => {
+  let { value } = props;
+  const { controlProps } = props;
+  const { handleChange, path, data } = controlProps;
+
+  if (value && value !== null) {
+    value = standardizeDate(value) || '';
+    if (value !== data) {
+      handleChange(path, value);
+    }
+  }
+};
+
+/**
+ * Helper function to process onChange event for Date controls.
+ * @param props - EventChangeControlProps
+ */
+export const onChangeForDateTimeControl = (props: EventChangeControlProps) => {
+  let { value } = props;
+  const { controlProps } = props;
+  const { handleChange, path, data } = controlProps;
+
+  if (value && value !== null) {
+    value = isValidDate(value) ? new Date(value)?.toISOString() : '';
+    if (data !== value) {
+      handleChange(path, value);
+    }
+  }
+};
+
+/**
+ * Helper function to process onChange event for Number/Integer controls.
+ * @param props - EventChangeControlProps
+ */
+export const onChangeForNumericControl = (props: EventChangeControlProps) => {
+  const { value } = props;
+  const { controlProps } = props;
+  const { handleChange, path, data } = controlProps;
+
+  if (value && value !== null) {
+    //Prevents handleChange from executing if the data has not changed
+    //so the component will not re render.
+    if (data !== +value) {
+      let newValue: string | number = '';
+      if (value !== '') {
+        newValue = +value;
+      }
+      handleChange(path, newValue);
+    }
+  }
+};
```

Comment:
Moving these things into utilities is a great idea; less clutter in the other files.  One suggestion though: there seems to be a fair amount of duplicate code.  One way to manage this is to have base functions that you pass a function into, so something like:
```
const onKeyPress = (props: EventKeyPressControlProps, handler: () => void) => {
  return (props) => {
    const { controlProps, key, value } = props;
    const { path } = controlProps;

    if (isNotKeyPressTabOrShift(key)) {
      handler(path, value);
    }
  }
};
```
and then, for example:
```
const onKeyPressForTimeControl = onKeyPress(props: EventKeyPressControlProps, props.handleChange)
```
The reason its only a suggestion is that I'm not sure if it would clean things up sufficiently to be worth while.